go/wallet/wallet.go

```diff
@@ -29,6 +31,12 @@ type Wallet interface {
 
 	// PrivateKey returns the wallets private key
 	PrivateKey() *ecdsa.PrivateKey
+
+	// SetViewingPrivKey and GetViewingPrivKey and wallet can cache its current registered viewing private key here.
+	//
+	// In testing we may need to set VK on the rpc client before each request as clients are shared between many wallets
+	SetViewingPrivKey(viewingPrivKey *ecies.PrivateKey)
```

Comment:
@BedrockSquirrel Can we add a todo to remove these methods? I'm imagining that once we get to having viewing keys in the sims, we'll need to have a dedicated L2 client per wallet (this is a more accurate representation of real-world usage anyway), in which case I don't believe we'll need this stuff?